hudi-client/hudi-client-common/src/main/java/org/apache/hudi/client/transaction/lock/FileSystemBasedLockProvider.java

```diff
@@ -87,8 +87,13 @@ public boolean tryLock(long time, TimeUnit unit) {
     try {
       synchronized (LOCK_FILE_NAME) {
         // Check whether lock is already expired, if so try to delete lock file
-        if (fs.exists(this.lockFile) && checkIfExpired()) {
-          fs.delete(this.lockFile, true);
+        if (fs.exists(this.lockFile)) {
+          if (checkIfExpired()) {
+            fs.delete(this.lockFile, true);
+            LOG.warn("Delete expired lock file: " + this.lockFile);
+          } else {
+            return false;
```

Comment:
Let's cover this scenario in `TestFileBasedLockProvider`?

Reply:
It should be covered by `#testReentrantLock`